django/db/models/base.py

```diff
@@ -489,9 +489,8 @@ def __init__(self, *args, **kwargs):
     @classmethod
     def from_db(cls, db, field_names, values):
         if len(values) != len(cls._meta.concrete_fields):
-            values = list(values)
-            values.reverse()
-            values = [values.pop() if f.attname in field_names else DEFERRED for f in cls._meta.concrete_fields]
+            values_iter = iter(values)
+            values = [next(values_iter) if f.attname in field_names else DEFERRED for f in cls._meta.concrete_fields]
```

Comment:
I think multilining would improve readability:
```
values = [
    next(values_iter) if f.attname in field_names else DEFERRED
    for f in cls._meta.concrete_fields
]
```
(in that case can the `values_iter` variable be removed?)